### ISSUE TITLE
command to download all the courses the user can see

### ISSUE THREAD

**Issue by sds100** [C0D3D3V/Moodle_Downloader_2#6]
Could you add a command to download all the courses the user can SEE and not just ones they are enrolled in since I can see other courses on my moodle, which I'm not enrolled in?

**Comment by C0D3D3V** [comment]
You mean like a white list? 
So that only the courses on that list are downloaded? Currently there is only a blacklist implemented. 

In the issue #4  some ideas were mentioned that I still have to think about, so I have not implemented a whitelist yet. It is expected that soon it will be possible to define more precisely which courses should be downloaded (also that you can rename them ect.)

But if you want to try something on it, you are welcome to try to implement it yourself, you can use the blacklist for orientation.

**Comment by C0D3D3V** [comment]
That would be fine, but it is very unusual to download courses in which you are not enrolled. 

Especially because by default you can't see the course content without registering, the admin has to change some settings to make this possible.

Currently I don't have time to implement this, but you can currently try to force the downloader to download a course by modifying the following code.
In this file moodle_connector/moodle_service.py

There, in line 90, all courses you are enrolled in are selected.
`courses = results_handler.fetch_courses(userid)`

You can add courses manually immediately afterwards.

Like this:
```
courses = results_handler.fetch_courses(userid)
courses.append(Course(2342, 'Ein Kurs', []))
```
Where the first parameter of Course is the id of the course, the second the name of the course and the 3rd must be an empty list.

This should work as long as you have read access to the course.

**Comment by sds100** [comment]
Yeah it is unusual. ?? Thank you for the instructions! I'll have a go myself.
________________________________

**Comment by C0D3D3V** [comment]
You can have a look in `results_handler.fetch_courses` and make a similar function for all courses...

You get a list of all courses in your moodle by calling:

https://moodle.uni.de/m/webservice/rest/server.php?moodlewsrestformat=json&wsfunction=core_course_get_courses_by_field&wstoken=c3703dcyour_token2e2eb46ba942a0

If there are a lot of courses on your moodle, it will not make you happy... because it takes a very long time.

To filter the list, you should first ask for the catagories... and then filter by category.

https://moodle.uni.de/m/webservice/rest/server.php?moodlewsrestformat=json&wsfunction=core_course_get_categories&wstoken=c3703dcyour_token2e2eb46ba942a0

https://moodle.uni.de/m/webservice/rest/server.php?moodlewsrestformat=json&wsfunction=core_course_get_courses_by_field&wstoken=c3703dcyour_token2e2eb46ba942a0&field=category&value=437

**Comment by C0D3D3V** [comment]
First steps have been taken for this issue :D 
Check out Issue https://github.com/C0D3D3V/Moodle-Downloader-2/issues/71 for related work.

**Comment by sds100** [comment]
Cool! 😀

**Comment by C0D3D3V** [comment]
you can now use `moodle-dl --add-all-visible-courses` 
It will be in the next release, you are welcome :)
